spec/importers/reading/reading_processor_spec.rb

```diff
@@ -0,0 +1,67 @@
+require 'rails_helper'
+
+RSpec.describe MegaReadingProcessor do
+  def fixture_file_text
+    IO.read("#{Rails.root}/spec/importers/reading/reading_processor.csv")
+  end
+
+  def create_test_students(pals)
+    Student.create!(
+      first_name: 'Pluto',
+      last_name: 'Skywalker',
+      school: pals.healey,
+      "grade": '2',
+      local_id: '2',
+      enrollment_status: 'Active'
+    )
+    Student.create!(
+      first_name: 'Donald',
+      last_name: 'Skywalker',
+      school: pals.healey,
+      "grade": '2',
+      local_id: '3',
+      enrollment_status: 'Active'
+    )
+  end
+
+  describe 'integration test' do
+    it 'works on happy path' do
+      pals = TestPals.create!
+      students = create_test_students(pals)
+      processor = MegaReadingProcessor.new(pals.uri.id, header_rows_count: 3)
+      reading_data = processor.process(fixture_file_text)
+      puts reading_data[0].as_json(except: [:student_id, :imported_by_educator_id])
+
+      expect(reading_data.size).to eq 2
+      expect(reading_data[0].as_json(except: [:student_id, :imported_by_educator_id])).to eq([
```

Comment:
These tests are great!  Can you update to also verify the student is correct?  This would be a major problem if it were not :)  Since the student record will have a different id reach run, you can store the record in a variable, and then reference its id in the test assertion (eg, `donald = Student.create!(..)` and then later check that `student_id: donald.id`)

Reply:
addressed in https://github.com/studentinsights/studentinsights/pull/2544/files#diff-f9626d0b749afc86a13a97049c70cceaR36